arches/app/media/js/views/resource/new-editor.js

```diff
@@ -145,6 +147,27 @@ define([
         reviewer: data.userisreviewer,
         graphiconclass: data.graphiconclass,
         relationship_types: data.relationship_types,
+        appliedFunctions: appliedFunctions(),
+        isFuncNode: function() {
+            var node = false;
+            var appFuncDesc, appFuncName;
+            if(this.appliedFunctions.length > 0) {
+                appFuncDesc = this.appliedFunctions[0]['config']['description']['nodegroup_id'];
```

Comment:
this section has the same problems as described above re: isFuncNode in card.js, ie keying on properties that may not be present and assuming the function of interest is at index 0 of the array.

Reply:
Does this code really need to be duplicated in so many places? can't we just always use `isFuncNode` on the card viewmodel?

Reply:
it looks like isFuncNode in new-editor.js wasn't actually being used, so I nixed that one. Regarding just having one isFuncNode method in the card viewmodel and calling to that from the other viewmodels, I tried to assign something like a "this.isFuncNode: CardViewModel.isFuncNode" (with and w/o parentheses) and get "not a function" errors. The other thing I wasn't sure about was the fact that I'm relying on different inputs like `params`,`self`, and `node` and it wasn't clear how to reconcile that if I was just going to export that one function to the other viewmodels.